plugin/src/main/kotlin/com/github/triplet/gradle/play/PlayPublisherPlugin.kt

```diff
@@ -97,13 +97,24 @@ class PlayPublisherPlugin : Plugin<Project> {
             }
 
             if (variant.isSigningReady) {
+                val processPackageMetadata = project.newTask<ProcessPackageMetadataTask>(
+                        "processPackageMetadata$variantName",
+                        "Processes packaging metadata for $variantName.",
+                        null
+                ) {
+                    init()
+
+                    variant.checkManifest.dependsOn(this)
```

Comment:
I am highly concerned about this task dependency. Since the `check[Variant]Manifest` is run very early in the build chain, we would introduce a call to the play store API even though in most cases we don't have to. E.g. check this tree:

```
--- :app:generateReleaseBuildConfig
     \--- :app:checkReleaseManifest
          \--- :app:preReleaseBuild
               \--- :app:preBuild
```

generating the build config is usually even triggered by Android Studio after syncing the gradle scripts.

Could we find a way to run this task only when absolutely necessary?

Reply:
Yeah, I know. Sadly, there's no way around this because once the manifest task is run, our version code overrides are ignored. My previous implementation was even worse, it queried stuff at configuration time! 🤦‍♂️ Here's why this isn't such a big deal: our tasks are only applicable for release builds which means normal development won't be affected.

Either way, I think your fear is well founded and I'll short circuit the query for `IGNORED` so the default isn't affected.